### PR TITLE
CORS-3594: Setting CAPG as the default infra provider

### PR DESCRIPTION
**Pull request by barbacbd**
** CAPG should be used as the default infra provider for GCP installs.

### CODE REVIEW

File: features/features.go [hunk 1]
```diff
@@ -575,9 +575,9 @@ var (
...
-					reportProblemsToJiraComponent("Routing").
-					contactPerson("miciah").
-					productScope(ocpSpecific).
-					enableIn(configv1.DevPreviewNoUpgrade, configv1.TechPreviewNoUpgrade).
-					mustRegister()
+							reportProblemsToJiraComponent("Routing").
+							contactPerson("miciah").
+							productScope(ocpSpecific).
+							enableIn(configv1.DevPreviewNoUpgrade, configv1.TechPreviewNoUpgrade).
+							mustRegister()
```
Comment (r4f4) [inline features/features.go:582]: I guess it's from your editor's auto-formatting, you can drop these changes.

**Review by r4f4 — COMMENTED** [review]
lgtm just left a nit comment.

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Comment by patrickdillon** [comment]
#1909 fixed the tests, which are now failing with:

```
 INSUFFICIENT CI testing for "ClusterAPIInstallGCP".
F0715 17:49:34.051041  169158 root.go:64] Error running codegen: error: "install should succeed: infrastructure" only passed 71%, need at least 95% for "ClusterAPIInstallGCP" on {gcp amd64 ha} 
```

The figure 71% seems off to me. That is, I don't think the infrastructure provisioning success rate is that low. I'm not sure where the discrepancy is coming from.

I'm reviewing the GCP Tech preview installs here: https://sippy.dptools.openshift.org/sippy-ng/jobs/4.17/runs?filters=%7B%22items%22%3A%5B%7B%22columnField%22%3A%22name%22%2C%22operatorValue%22%3A%22equals%22%2C%22value%22%3A%22periodic-ci-openshift-release-master-ci-4.17-e2e-gcp-ovn-techpreview%22%7D%5D%7D&pageSize=100&sort=desc&sortField=timestamp

Reviewing these failures, the significant one I see is the credentials request failure which recurs multiple times, including this example: https://prow.ci.openshift.org/view/gs/test-platform-results/logs/periodic-ci-openshift-release-master-ci-4.17-e2e-gcp-ovn-techpreview/1806798263932686336

That issue was not related to `ClusterAPIInstallGCP` and was fixed in: https://issues.redhat.com/browse/OCPBUGS-36294

The only issue I see related to `ClusterAPIInstallGCP` is 

```
level=error msg=failed to fetch Cluster: failed to generate asset "Cluster": failed to create cluster: failed during pre-provisioning: failed to add worker roles: failed to set project IAM policy: googleapi: Error 409: There were concurrent policy changes. Please retry the whole read-modify-write with exponential backoff. The request's ETag '\007\006\033\255\347+\335\210' did not match the current policy's ETag '\007\006\033\255\347>%\332'., aborted
Installer 
```
from: https://prow.ci.openshift.org/view/gs/test-platform-results/logs/periodic-ci-openshift-release-master-ci-4.17-e2e-gcp-ovn-techpreview/1805429311365189632

That is something we'll want to fix and would potentially be fixed by https://issues.redhat.com/browse/CORS-3567

**Comment by stbenjam** [comment]
> The figure 71% seems off to me. That is, I don't think the infrastructure provisioning success rate is that low. I'm not sure where the discrepancy is coming from.

I'm looking to figure out where the 71% number came from but techpreview gcp infra is low.  The default sippy view is "Working" which is flake + success.  For this we're using success only.

Sippy is currently saying 89% in the UI:

https://sippy.dptools.openshift.org/sippy-ng/tests/4.17/details?filters=%257B%2522items%2522%253A%255B%257B%2522columnField%2522%253A%2522current_runs%2522%252C%2522operatorValue%2522%253A%2522%253E%253D%2522%252C%2522value%2522%253A%25227%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Afalse%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Platform%253Agcp%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Afalse%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522FeatureSet%253Atechpreview%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Topology%253Aha%2522%257D%252C%257B%2522id%2522%253A99%252C%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522infrastructure%2522%257D%252C%257B%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522install%2520should%2522%257D%255D%252C%2522linkOperator%2522%253A%2522and%2522%257D&period=default&sort=asc&sortField=current_pass_percentage&view=Passing

**Comment by stbenjam** [comment]
Non-techpreview GCP definitely has much higher infra success https://sippy.dptools.openshift.org/sippy-ng/tests/4.17/details?filters=%257B%2522items%2522%253A%255B%257B%2522columnField%2522%253A%2522current_runs%2522%252C%2522operatorValue%2522%253A%2522%253E%253D%2522%252C%2522value%2522%253A%25227%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Afalse%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Platform%253Agcp%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Atrue%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522FeatureSet%253Atechpreview%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Topology%253Aha%2522%257D%252C%257B%2522id%2522%253A99%252C%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522infrastructure%2522%257D%252C%257B%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522install%2520should%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Architecture%253Aamd64%2522%257D%255D%252C%2522linkOperator%2522%253A%2522and%2522%257D&period=default&sort=asc&sortField=current_pass_percentage&view=Passing

**Comment by patrickdillon** [comment]
Just reran verify and it looks like our bug fixes are paying off and we're trending in the right direction (86%, up from 71%):

```
 F0729 19:20:08.510204  169977 root.go:64] Error running codegen: error: "install should succeed: infrastructure" only passed 86%, need at least 95% for "ClusterAPIInstallGCP" on {gcp amd64 ha} 
```

**Comment by patrickdillon** [comment]
I ran ~20 GCP techpreview jobs yesterday using gangway. Looking at the infrastructure test links that @stbenjam posted above, I believe we are now seeing a success rate ~98%:

[GCP TechPreview Infrastructure](https://sippy.dptools.openshift.org/sippy-ng/tests/4.17/details?filters=%257B%2522items%2522%253A%255B%257B%2522columnField%2522%253A%2522current_runs%2522%252C%2522operatorValue%2522%253A%2522%253E%253D%2522%252C%2522value%2522%253A%25227%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Afalse%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Platform%253Agcp%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Afalse%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522FeatureSet%253Atechpreview%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Topology%253Aha%2522%257D%252C%257B%2522id%2522%253A99%252C%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522infrastructure%2522%257D%252C%257B%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522install%2520should%2522%257D%255D%252C%2522linkOperator%2522%253A%2522and%2522%257D&period=default&sort=asc&sortField=current_pass_percentage&view=Passing)

This seems to be actually higher than the non-tech preview tests, which are at around 96-97%:

[Non-tech preview](https://sippy.dptools.openshift.org/sippy-ng/tests/4.17/details?filters=%257B%2522items%2522%253A%255B%257B%2522columnField%2522%253A%2522current_runs%2522%252C%2522operatorValue%2522%253A%2522%253E%253D%2522%252C%2522value%2522%253A%25227%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Afalse%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Platform%253Agcp%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Atrue%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522FeatureSet%253Atechpreview%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Topology%253Aha%2522%257D%252C%257B%2522id%2522%253A99%252C%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522infrastructure%2522%257D%252C%257B%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522install%2520should%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Architecture%253Aamd64%2522%257D%255D%252C%2522linkOperator%2522%253A%2522and%2522%257D&period=default&sort=asc&sortField=current_pass_percentage&view=Passing)

In other words, despite verify test failures, this is looking good to me in regards to CI testing.
